francis/noise_reduction.py

```diff
@@ -0,0 +1,44 @@
+import numpy as np
+import librosa
+import noisereduce
+
+
+def process(audio_data, seconds=2):
+    return noisereduce.reduce_noise(
+        audio_clip=audio_data[0],
+        noise_clip=__backround_noise(audio_data, seconds),
+        verbose=False,
+    )
+
+
+def __backround_noise(audio_data, seconds):
+    return __quietest(__split_audio(audio_data, seconds))
+
+
+def __quietest(split_amplitudes):
+    quietest_volume = None
+    quietest_index = 0
+    for i in range(len(split_amplitudes)):
+        average_volume = np.mean(np.abs(split_amplitudes[i]))
+        if quietest_volume == None or quietest_volume > average_volume:
+            quitest_volume = average_volume
+            quietest_index = i
+    return split_amplitudes[quietest_index]
+
+
+def __split_audio(audio_data, seconds):
+    amplitudes, rate = audio_data
+    size_of_elements = rate * seconds
+    number_of_elements = len(amplitudes) // size_of_elements
+    block = round(size_of_elements * number_of_elements)
+    divisible_amplitudes = amplitudes[:block]
+    leftover_amplitudes = amplitudes[block:]
+
+    return __append_leftover(
+        np.split(divisible_amplitudes, number_of_elements), leftover_amplitudes
+    )
+
```

Comment:
Here one audio file is taken, and it split it into x seconds chunks, 
the left over audio is added to the last chunk.
a nested numpy array is returned in this format
`[[audio_data_a-b_seconds],[audio_data_b-c_seconds], ....]`